xarray_beam/_src/core.py

```diff
@@ -340,12 +340,14 @@ def _validate(self, dataset, split_vars):
         raise ValueError("dataset list cannot be empty")
     sizes = [ds.sizes for ds in self._datasets]
     if len({tuple(s.items()) for s in sizes}) > 1:
-      raise ValueError(f"inconsistent dataset sizes: {sizes}")
+      if any(len(sizes[0]) < len(s) for s in sizes[1:]):
+        raise ValueError(f"inconsistent dataset sizes, only the first can be greater: {sizes}")
```

Comment:
I'm finding it a little hard to follow these checks written in terms of sizes. I'm also not sure they're right, because you could have a case where the second dataset has fewer dimensions but they are new dimensions.

How about something like this instead?
```python
for ds in self._datasets[1:]:
    for dim, size in ds.sizes.items():
        if dim not in self._first.dims:
            raise ValueError(f"dimension {dim} does not appear on the first dataset")
        if size != self._first.sizes[dim]:
            raise ValueError(f"dimension {dim} has an inconsistent size on different datasets")
```